Assets/Scripts/Analytics/TimingEventAnalytics.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using Unity.Services.Analytics;
+using UnityEngine;
+
+namespace Analytics
+{
+    /// <summary>
+    /// Script used to time a specific event.
+    /// Trigger StartEvent to start the timing, and EndEvent to end timing.
+    /// Use a new instance of this script for each event to be timed.
+    /// </summary>
+    public class TimingEventAnalytics : MonoBehaviour
+    {
+        [SerializeField] private string analyticStartedName;
+        [SerializeField] private string analyticEndedName;
+        [SerializeField] private string analyticEndTimeName;
+        [SerializeField] private bool sendStartEvent = true;
+
+        private float _cachedTimeCreditsOpen;
```

Comment:
There's a few references to 'credits' in this file, but the actual code seems quite generic, is that intentional?

Reply:
It is not - good catch :D